Fix a very sporadic issue of a wrong comment header applied

okurz opened the pull request:
From time to time our CI tests show a problem like the following:

```
not ok 4 - heading text
```

I managed to reproduce it locally although this was really hard.  One of two
runs of

```
STABILITY_TEST=1 RETRY=200 tools/retry prove -v -l t/ui/15-comments.t
```

could reproduce it in "Rerun 125 of 200 …" for the first time.

The problem most likely appears in javascript when we want to update the
comment header line but there is an incorrect "edited" appended as the creation
and update time might not be set to exactly the same in the database row.

To verify a fix I ran:

```
STABILITY_TEST=1 RETRY=500 tools/retry prove -l t/ui/15-comments.t
```

successfully with no failed tests.

Related progress issue: https://progress.opensuse.org/issues/53771

okurz commented:
What @Martchus wrote in another channel:

> It is not nice that this can happen in the first place but I agree that adding 
> this workaround is likely easier than ensuring that t_created and t_updated are 
> always the same when a row has just been created.
>
> The absence of a unit in your comment made me curious what the 1 actually means 
> here. It turns out that these timestamps are strings like "2020-04-23 13:13:31 > +0000" here. So you're basically appending to the timezone. It seems "to do the > trick" but doesn't look like a good idea to me. You could use the following
> instead:
> if((new Date(comment.updated).getTime() - 1000) >= (new Date(comment.created))) { …
> 
> But nevertheless it is nice that the issue is finally tackled (even if just worked around).

Decreasing the value by 1000 does not seem clearer to me though.

Maybe we can find a proper period during which we would not consider an update an "edit" anyway?

Martchus commented:
I *assumed* the "1" in your comment is intended to be one second. So for consistency with your own version I've used 1000 here as `getTime()` returns milliseconds. But we could also generally put a higher number here. I don't care much about it.

okurz commented:
changed

```diff
-    // add a margin of 1 to prevent erroneous update detected if off by 1
-    if(comment.updated > (comment.created + 1)) {
+    // add a margin of 1000ms to prevent erroneous update detected if off by 1
+    if((new Date(comment.updated).getTime() - 1000) >= (newDate(comment.created))) {
```

I have not tested it 500-600 times as that would take another 24h :D
But it passed at least 6 runs locally so far ;)
